Description: Update to comply with the latest FFmpeg API.
Author: Alessio Treglia <quadrispro@ubuntu.com>
Ubuntu-Bug: https://launchpad.net/bugs/646824
---
 Makefile.am     |    3 +--
 configure.in    |    3 +--
 src/xtoffmpeg.c |   14 +++++++-------
 3 files changed, 9 insertions(+), 11 deletions(-)

--- xvidcap-1.1.7.orig/src/xtoffmpeg.c
+++ xvidcap-1.1.7/src/xtoffmpeg.c
@@ -95,7 +95,7 @@
                         || (x)==PIX_FMT_YUVJ420P || (x)==PIX_FMT_YUVJ422P \
                         || (x)==PIX_FMT_YUVJ444P)
 
-#define PIX_FMT_ARGB32 PIX_FMT_RGBA32  /* this is just my personal
+#define PIX_FMT_ARGB32 PIX_FMT_RGB32  /* this is just my personal
                                         * convenience */
 
 /*
@@ -194,7 +194,7 @@ typedef struct AVOutputStream
     /* audio only */
     int audio_resample;
     ReSampleContext *resample;  /* for audio resampling */
-    AVFifoBuffer fifo;  /* for compression: one audio fifo per
+    AVFifoBuffer *fifo;  /* for compression: one audio fifo per
                          * codec */
     FILE *logfile;
 } AVOutputStream;
@@ -390,7 +390,7 @@ add_audio_stream (Job * job)
     }
     au_out_st->st->codec = au_c;
 
-    if (av_fifo_init (&au_out_st->fifo, 2 * MAX_AUDIO_PACKET_SIZE)) {
+    if ( (au_out_st->fifo = av_fifo_alloc (2 * MAX_AUDIO_PACKET_SIZE)) ) {
         fprintf (stderr,
                  _("%s %s: Can't initialize fifo for audio recording\n"),
                  DEBUGFILE, DEBUGFUNCTION);
@@ -548,10 +548,10 @@ do_audio_out (AVFormatContext * s, AVOut
     // now encode as many frames as possible
     if (enc->frame_size > 1) {
         // output resampled raw samples
-        av_fifo_write (&ost->fifo, buftmp, size_out);
+        av_fifo_generic_write (ost->fifo, buftmp, size_out, NULL);
         frame_bytes = enc->frame_size * 2 * enc->channels;
 
-        while (av_fifo_read (&ost->fifo, audio_buf, frame_bytes) == 0) {
+        while (av_fifo_generic_read (ost->fifo, audio_buf, frame_bytes, NULL) == 0) {
             AVPacket pkt;
 
             // initialize audio output packet
@@ -659,7 +659,7 @@ cleanup_thread_when_stopped ()
     enc = au_out_st->st->codec;
     samples = av_malloc (AVCODEC_MAX_AUDIO_FRAME_SIZE);
     bit_buffer = av_malloc (bit_buffer_size);
-    fifo_bytes = av_fifo_size (&au_out_st->fifo);
+    fifo_bytes = av_fifo_size (au_out_st->fifo);
     ret = 0;
 
     /* encode any samples remaining in fifo */
@@ -669,7 +669,7 @@ cleanup_thread_when_stopped ()
         int fs_tmp = enc->frame_size;
 
         enc->frame_size = fifo_bytes / (2 * enc->channels);
-        if (av_fifo_read (&au_out_st->fifo, (uint8_t *) samples, fifo_bytes) ==
+        if (av_fifo_generic_read (au_out_st->fifo, (uint8_t *) samples, fifo_bytes, NULL) ==
             0) {
             ret =
                 avcodec_encode_audio (enc, bit_buffer, bit_buffer_size,
